src/network/networkd-link.c

```diff
@@ -284,8 +299,9 @@ static int link_enable_ipv6(Link *link) {
         return 0;
 }
 
-void link_update_operstate(Link *link) {
+void link_update_operstate(Link *link, bool also_update_bond_master) {
```

Comment:
Why the bool required ?

Reply:
When this is called for bonding master, then the states of its slaves should be also updated.
Without this flag, master updates its slaves, and slaves update their master, and... infinite loop.